packages/wrangler/src/publish.ts

```diff
@@ -288,6 +289,7 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 
 	const jsxFactory = props.jsxFactory || config.jsx_factory;
 	const jsxFragment = props.jsxFragment || config.jsx_fragment;
+	const keepVars = props.keepVars || config.keep_vars;
```

Comment:
```suggestion
	const keepVars = props.keepVars ?? config.keep_vars;
```

That way `--keep-vars=false` would override `keep_vars = true` in wrangler.toml.